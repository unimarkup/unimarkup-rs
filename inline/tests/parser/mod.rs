use crate::snapshot::Snapshot;
use unimarkup_commons::test_runner::snap_test_runner::SnapTestRunner;
use unimarkup_inline::ParseInlines;

mod snapshot;

#[test]
fn test_parser_snapshots() {
    for case in crate::prepare_test_cases("spec/markup", "spec/snapshots/parser") {
        let runner = SnapTestRunner::with_fn(&case.name, case.input.as_str(), |symbols| {
            let rest: &[_] = &[];
            let inlines: Vec<_> = symbols.parse_inlines().collect();
            let snapshot = Snapshot::snap(&inlines[..]);
            (snapshot, rest)
        })
        .with_info(format!("Test '{}' from '{}'", case.name, case.file_name));

        unimarkup_commons::run_snap_test!(runner, &case.out_path);
    }
}
