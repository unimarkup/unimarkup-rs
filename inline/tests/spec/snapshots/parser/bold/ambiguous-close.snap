---
source: inline/tests/parser/mod.rs
info: "Test 'ambiguous-close' from 'markup\\bold.yml'"
---
Bold @ (1:1)->(1:19) (
    Plain @ (1:3)->(1:8) (
        bold 
        ^^^^^
    )
    Italic @ (1:8)->(1:17) (
        Plain @ (1:9)->(1:16) (
            +italic
            ^^^^^^^
        )
    )
)
Plain @ (1:19)->(1:25) (
     plain
    ^^^^^^
)

---
With input:

**bold *+italic*** plain


